### PR TITLE
AC-577 OpenMrs image do not cuts off in landscape mode

### PR DESCRIPTION
**Pull request by rishabh-997**
## Description of what I changed
1. I added the logo in scroll view
2.  removed the bug caused by merging of the previous PR where the openMRS logo was shown after fetching data and thus, after onResume, the logo was not visible as the function to show logo was not called in 5 cases

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-577

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by f4ww4z** [comment]
@rishabh-997 You need to pull the latest changes from master. Try `git pull --rebase upstream master` and fix any merge conflicts if any.

**Comment by rishabh-997** [comment]
> @rishabh-997 You need to pull the latest changes from master. Try `git pull --rebase upstream master` and fix any merge conflicts if any.

Nice... it completely solved the issue of extra commits as well.

File: openmrs-client/src/main/res/layout/fragment_login.xml [hunk 1]
```diff
@@ -12,204 +12,201 @@
...
+
+			<ProgressBar
+				android:id="@+id/loginLoading"
+				android:layout_width="wrap_content"
+				android:layout_height="wrap_content"
+				android:layout_gravity="center"
+				android:layout_marginBottom="10dp" />
+		</LinearLayout>
+	</RelativeLayout>
+</ScrollView>
```
Comment (Devanshk9) [inline openmrs-client/src/main/res/layout/fragment_login.xml]: Hi @rishabh-997 
Kindly leave a new line at the end.

File: openmrs-client/src/main/res/layout/fragment_login.xml [hunk 2]
```diff
@@ -12,204 +12,201 @@
...
-
-            <ProgressBar
-                android:id="@+id/loginLoading"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:layout_gravity="center"
-                android:layout_marginBottom="10dp" />
-        </LinearLayout>
-    </ScrollView>
-</RelativeLayout>
```
Comment (Devanshk9) [inline openmrs-client/src/main/res/layout/fragment_login.xml]: @rishabh-997 same here

Comment (rishabh-997) [reply openmrs-client/src/main/res/layout/fragment_login.xml]: these lines do not exist...

File: openmrs-client/src/main/res/layout/fragment_login.xml (hunk 1 above)
Comment (rishabh-997) [reply openmrs-client/src/main/res/layout/fragment_login.xml]: [Have a look at my comment](https://github.com/openmrs/openmrs-contrib-android-client/pull/650#discussion_r364606911)

**Comment by f4ww4z** [comment]
@rishabh-997 I've made a commit for the release v2.8.1, can you pull from master again please.

**Comment by rishabh-997** [comment]
> @rishabh-997 I've made a commit for the release v2.8.1, can you pull from master again please. Also don't worry about the previous comments.

You can merge now @f4ww4z  sir

**Comment by f4ww4z** [comment]
@rishabh-997 There's this error when I run it:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.openmrs.mobile/org.openmrs.mobile.activities.login.LoginActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.util.SparseArray.get(int)' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2895)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4766)
        at android.app.ActivityThread.-wrap18(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1624)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:176)
        at android.app.ActivityThread.main(ActivityThread.java:6651)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.util.SparseArray.get(int)' on a null object reference
        at org.openmrs.mobile.activities.login.LoginFragment.createImageBitmap(LoginFragment.java:311)
        at org.openmrs.mobile.activities.login.LoginFragment.showOpenMRSLogo(LoginFragment.java:439)
        at org.openmrs.mobile.activities.login.LoginFragment.initLoginForm(LoginFragment.java:337)
        at org.openmrs.mobile.activities.login.LoginFragment.hideURLDialog(LoginFragment.java:433)
        at org.openmrs.mobile.activities.login.LoginFragment.onCreateView(LoginFragment.java:105)
        at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2600)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:881)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2659)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2613)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246)
...
```

[See this recording](https://imgur.com/a/HTeenu9). Can you fix it?

**Comment by rishabh-997** [comment]
> @rishabh-997 There's this error when I run it:
> 
> ```
> java.lang.RuntimeException: Unable to start activity ComponentInfo{org.openmrs.mobile/org.openmrs.mobile.activities.login.LoginActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.util.SparseArray.get(int)' on a null object reference
>         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
>         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2895)
>         at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4766)
>         at android.app.ActivityThread.-wrap18(Unknown Source:0)
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1624)
>         at android.os.Handler.dispatchMessage(Handler.java:106)
>         at android.os.Looper.loop(Looper.java:176)
>         at android.app.ActivityThread.main(ActivityThread.java:6651)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
>      Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.util.SparseArray.get(int)' on a null object reference
>         at org.openmrs.mobile.activities.login.LoginFragment.createImageBitmap(LoginFragment.java:311)
>         at org.openmrs.mobile.activities.login.LoginFragment.showOpenMRSLogo(LoginFragment.java:439)
>         at org.openmrs.mobile.activities.login.LoginFragment.initLoginForm(LoginFragment.java:337)
>         at org.openmrs.mobile.activities.login.LoginFragment.hideURLDialog(LoginFragment.java:433)
>         at org.openmrs.mobile.activities.login.LoginFragment.onCreateView(LoginFragment.java:105)
>         at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2600)
>         at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:881)
>         at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
>         at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
>         at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2659)
>         at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2613)
>         at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246)
> ...
> ```
> 
> [See this recording](https://imgur.com/a/HTeenu9). Can you fix it?

I got no such error while building... will see if I can replicate it and try to fix it...

**Comment by rishabh-997** [comment]
> @rishabh-997 There's this error when I run it:
> 
> ```
> java.lang.RuntimeException: Unable to start activity ComponentInfo{org.openmrs.mobile/org.openmrs.mobile.activities.login.LoginActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.util.SparseArray.get(int)' on a null object reference
>         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
>         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2895)
>         at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4766)
>         at android.app.ActivityThread.-wrap18(Unknown Source:0)
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1624)
>         at android.os.Handler.dispatchMessage(Handler.java:106)
>         at android.os.Looper.loop(Looper.java:176)
>         at android.app.ActivityThread.main(ActivityThread.java:6651)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
>      Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.util.SparseArray.get(int)' on a null object reference
>         at org.openmrs.mobile.activities.login.LoginFragment.createImageBitmap(LoginFragment.java:311)
>         at org.openmrs.mobile.activities.login.LoginFragment.showOpenMRSLogo(LoginFragment.java:439)
>         at org.openmrs.mobile.activities.login.LoginFragment.initLoginForm(LoginFragment.java:337)
>         at org.openmrs.mobile.activities.login.LoginFragment.hideURLDialog(LoginFragment.java:433)
>         at org.openmrs.mobile.activities.login.LoginFragment.onCreateView(LoginFragment.java:105)
>         at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2600)
>         at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:881)
>         at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
>         at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
>         at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2659)
>         at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2613)
>         at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246)
> ...
> ```
> 
> [See this recording](https://imgur.com/a/HTeenu9). Can you fix it?

@f4ww4z sir, i built the app and the login is working just fine... NO such error received at my end

**Comment by rishabh-997** [comment]
@f4ww4z sir, the error log you mentioned is similar to [AC-666](https://issues.openmrs.org/browse/AC-666). So you can merge this as of now and we can look into that issue separately.

**Comment by f4ww4z** [comment]
@rishabh-997 OK. Please solve merge conflicts and make sure its up to date with master branch. THen we can merge.

**Comment by rishabh-997** [comment]
> @rishabh-997 OK. Please solve merge conflicts and make sure its up to date with master branch. THen we can merge.

finally, ready for being merged @f4ww4z sir

**Comment by rishabh-997** [comment]
@f4ww4z  sir

**Pull request merged**